libraries/AP_Scripting/applets/Bendy_ruler.lua

```diff
@@ -0,0 +1,871 @@
+    local local_target_loc = Location()
+
+    --[[
+        fill in avoidance_result with current state
+    --]]
+    local now = millis()    -- Populate avoidance request with current data
+    avoidance_request.current_loc = current_loc
+    avoidance_request.target_loc = target_loc
+    avoidance_request.groundspeed = groundspeed    
+    -- Estimate airspeed (use groundspeed if no airspeed estimate available)
+    if not ahrs:airspeed_estimate() then
+        avoidance_request.airspeed = groundspeed
+    else
+        avoidance_request.airspeed = ahrs:airspeed_estimate()
+    end
+
+    -- Get wind estimate and convert to 2D
+    local wind_3d = ahrs:wind_estimate()
+    local wind_2d = Vector2f()
+    wind_2d:x(wind_3d:x())
+    wind_2d:y(wind_3d:y())
+
+    -- Calculate wind direction and speed
+    avoidance_request.wind_dir_rad = wind_2d:angle()
+    avoidance_request.wind_speed = wind_2d:length()    
+    -- Record request time
+    avoidance_request.request_time_ms = now    
+
+    -- call the main mission avoidance logic
+    local new_target_loc = update_mission_avoidance(avoidance_request, target_loc)
+
+    return new_target_loc
+end
+
+--[[
+    return true if two locations are identical
+--]]
+local function locations_equal(loc1, loc2)
+    return loc1:lat() == loc2:lat() and loc1:lng() == loc2:lng() and loc1:alt() == loc2:alt()
+end
+
+local orig_target_loc = nil
+local last_updated_target_loc = nil
+
+--[[
+    update avoidance at UPDATE_RATE_HZ. Calls mission_avoidance() and
+    updates target location in the vehicle code
+--]]
+function update_avoidance()
+    if AVD_ENABLE:get() == 0 then
+        -- disabled by the user
+        return
+    end
+
+    current_loc = ahrs:get_position()
+    target_loc = vehicle:get_target_location()
+
+    if not current_loc or not target_loc then
+        -- no position or not navigating
+        orig_target_loc = nil
+        last_updated_target_loc = nil
+        return
+    end
+
+    if not orig_target_loc or
+        not last_updated_target_loc or
+        not locations_equal(last_updated_target_loc, target_loc) then
+        -- the vehicle navigation code has changed it's target
+        orig_target_loc = target_loc:copy()
+        gcs:send_text(MAV_SEVERITY.INFO, "BR: new target loc")
+    end
+
+    local groundspeed = ahrs:groundspeed_vector():length()
+
+    -- run avoidance calculation
+    local new_target_loc = mission_avoidance(current_loc, orig_target_loc, groundspeed)
+
+    if new_target_loc then
+        -- tell the vehicle to fly to the new calculated target
+        vehicle:update_target_location(target_loc, new_target_loc)
+        last_updated_target_loc = new_target_loc:copy()
+
+        local avoid_dist = orig_target_loc:get_distance(new_target_loc)
+
+        gcs:send_named_float("AVD_DIST", avoid_dist)
+    else
+        -- revert to the original target location, no avoidance needed
+        vehicle:update_target_location(target_loc, orig_target_loc)
```

Comment:
It's not clear why you are doing this. It seems like it would already be going here, so why tell it to go where it's already going?